### ISSUE TITLE
cilium execute command

### ISSUE THREAD

**Issue by liyihuang** [cilium/cilium-cli#1930]
## Proposal / RFE

I personally have run these scenario a lot.

- I have a pod nginx running and I would like to get the cilium pod with this nginx. I will need to perform the following steps
    1. I run `kubectl get pods -o wide` to find out the node it's running for this nginx pod
    2. I run `kubectl get pods -l k8s-app=cilium -o name -n kube-system -o wid` to find out which cilium is on the same node with nginx
    3. I run `kubectl exec -it -n kube-system cilium-zmglg -- cilium xxx` or `kubectl exec -it -n kube-system cilium-zmglg -- bash` to get it cilium
 - I would like to collect all cilium commands for something, and I have to run these shell command
     1. `kubectl get pods -l k8s-app=cilium -o name -n kube-system | xargs -I {} sh -c 'echo Pod: {}; kubectl exec -n kube-system {} -- cilium status; echo'`

**Is your feature request related to a problem?**

**Describe the solution you'd like**

I hope we can have `cilium execute podname` where I will can run the cilium command to the pod that I want so I dont have to run step 1, 2, and 3 and `cilium execute all` where I can run the cilium command to all cilium pods

**Comment by liyihuang** [comment]
This is what it should look like. Let's say I want to run some command on cilium pod
```
cilium exec --pod nginx-xxx-yyy cilium-health status
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Probe time:   2023-08-23T14:32:36Z
Nodes:
  ip-10-2-6-163.us-east-2.compute.internal (localhost):
    Host connectivity to 10.2.6.163:
      ICMP to stack:   OK, RTT=404.483µs
      HTTP to agent:   OK, RTT=1.602324ms
    Endpoint connectivity to 10.2.6.107:
      ICMP to stack:   OK, RTT=412.573µs
      HTTP to agent:   OK, RTT=1.135271ms
  ip-10-2-5-115.us-east-2.compute.internal:
    Host connectivity to 10.2.5.115:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=2.150797ms
    Endpoint connectivity to 10.2.5.239:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=2.062174ms
  ip-10-2-6-21.us-east-2.compute.internal:
    Host connectivity to 10.2.6.21:
      ICMP to stack:   OK, RTT=1.118048ms
      HTTP to agent:   OK, RTT=944.225µs
    Endpoint connectivity to 10.2.6.132:
      ICMP to stack:   OK, RTT=396.655µs
      HTTP to agent:   OK, RTT=1.193838ms
  ip-10-2-6-49.us-east-2.compute.internal:
    Host connectivity to 10.2.6.49:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=3.801243ms
    Endpoint connectivity to 10.2.6.128:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=1.294341ms
  ip-10-2-7-128.us-east-2.compute.internal:
    Host connectivity to 10.2.7.128:
      ICMP to stack:   OK, RTT=1.112411ms
      HTTP to agent:   OK, RTT=1.402368ms
    Endpoint connectivity to 10.2.7.167:
      ICMP to stack:   OK, RTT=1.129288ms
      HTTP to agent:   OK, RTT=1.533837ms
  ip-10-2-7-36.us-east-2.compute.internal:
    Host connectivity to 10.2.7.36:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=1.357191ms
    Endpoint connectivity to 10.2.7.18:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=1.424669ms
  ip-10-2-8-160.us-east-2.compute.internal:
    Host connectivity to 10.2.8.160:
      ICMP to stack:   OK, RTT=1.831154ms
      HTTP to agent:   OK, RTT=2.174386ms
    Endpoint connectivity to 10.2.8.134:
      ICMP to stack:   OK, RTT=1.843083ms
      HTTP to agent:   OK, RTT=2.184002ms
```
run commands to all the pods

```
cilium exec --all cilium encrypt status
Pod: pod/cilium-5sv4z
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-cfmjz
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-dzp6w
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-m5qjn
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-qwj8b
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-qzpbg
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
Pod: pod/cilium-sxdwq
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), wait-for-node-init (init), clean-cilium-state (init), install-cni-binaries (init)
Encryption: Disabled
```

**Comment by joestringer** [comment]
For extra points, we should consider using [Ephemeral Containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) for this :)

**Comment by liyihuang** [comment]
@joestringer I just got some cycle to work on some ideas that I brought up last year. This is one of them. I dont understand how [Ephemeral Containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) could help. when I would like to debug cilium info for a nginx pod, and I need to run cilium command to access the cilium/BPF info. Then I always need 2-3 commands to find the cilium pod on the same node with the nginx pod. 

Hope I made myslef clear
